core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -684,22 +684,25 @@ class BlockManagerMasterEndpoint(
   private def getShufflePushMergerLocations(
       numMergersNeeded: Int,
       hostsToFilter: Set[String]): Seq[BlockManagerId] = {
-    val activeBlockManagers = blockManagerIdByExecutor.groupBy(_._2.host)
+    val blockManagersWithExecutors = blockManagerIdByExecutor.groupBy(_._2.host)
       .mapValues(_.head).values.map(_._2).toSet
-    val filteredActiveBlockManagers = activeBlockManagers
+    val filteredBlockManagersWithExecutors = blockManagersWithExecutors
       .filterNot(x => hostsToFilter.contains(x.host))
-    val filteredActiveMergers = filteredActiveBlockManagers.map(
+    val filteredMergersWithExecutors = filteredBlockManagersWithExecutors.map(
       x => BlockManagerId(x.executorId, x.host, StorageUtils.externalShuffleServicePort(conf)))
 
     // Enough mergers are available as part of active executors list
-    if (filteredActiveMergers.size >= numMergersNeeded) {
-      filteredActiveMergers.toSeq
+    if (filteredMergersWithExecutors.size >= numMergersNeeded) {
+      filteredMergersWithExecutors.toSeq
     } else {
       // Delta mergers added from inactive mergers list to the active mergers list
-      val filteredDeadMergers = shuffleMergerLocations.values
-        .filterNot(mergerHost => filteredActiveMergers.exists(x => x.host == mergerHost.host))
-      filteredActiveMergers.toSeq ++
-        filteredDeadMergers.toSeq.take(numMergersNeeded - filteredActiveMergers.size)
+      val filteredMergersWithExecutorsHosts = filteredMergersWithExecutors.map(_.host)
```

Comment:
This would be an Iterable right ? Convert it to Set with `toSet` ? Else the cost remains O(N^2)

Reply:
Scratch that, did not see the .toSet while creating `blockManagersWithExecutors` !

Reply:
I think it would be a `Set` as `filteredMergersWithExecutors` is already a `Set`. Also double checked the type and it is `Set[String` for `filteredMergersWithExecutorsHosts`

Reply:
Yeah, I missed that - hence my second followup comment :-)